src/Stratis.Bitcoin.Features.SmartContracts/Rules/P2PKHNotContractRule.cs

```diff
@@ -12,15 +13,20 @@ namespace Stratis.Bitcoin.Features.SmartContracts.Rules
     public class P2PKHNotContractRule : FullValidationConsensusRule
     {
         private readonly IStateRepositoryRoot stateRepositoryRoot;
+        private readonly int? lastCheckPointHeight;
 
-        public P2PKHNotContractRule(IStateRepositoryRoot stateRepositoryRoot)
+        public P2PKHNotContractRule(IStateRepositoryRoot stateRepositoryRoot, Network network)
         {
             this.stateRepositoryRoot = stateRepositoryRoot;
+            this.lastCheckPointHeight = network.Checkpoints.OrderByDescending(c => c.Key).Select(c => c.Key).FirstOrDefault();
         }
 
         /// <inheritdoc/>
         public override Task RunAsync(RuleContext context)
         {
+            if (context.ValidationContext.ChainedHeaderToValidate.Height <= this.lastCheckPointHeight)
```

Comment:
Similarly to my comment on #931, using the `SkipValidation` flag will more closely align this rule with how the others work.

Reply:
@rowandh do you think it is safe to skip `stateRepositoryRoot.GetAccountState(to)` in `CheckTransaction` below? I suppose at worst the account would get populated in the cache when it appears later on in the chain, but it is an area of the code I am less familiar with